File: BOJ/python3/2003.py
'''
수들의 합 2
https://www.acmicpc.net/problem/2003
'''
import sys
import time
start = time.time()  # 시작 시간
sys.stdin = open("input.txt", "r")  # input.txt 파일로 입력

N, M = map(int, input().split())
nums = list(map(int, input().split()))

ans = 0  # 합이 M이 되는 경우의 수
i, j = 0, 0  # low, high pointer
s = 0  # 구간의 합
while True:
    if s >= M:
        s -= nums[i]
        i += 1
    elif j == N:
        break
    else:
        s += nums[j]
        j += 1

    if s == M:
        ans += 1

print(ans)
# 현재시각 - 시작시간 = 실행 시간(sec)
print("--------------------")
print(f"Elapsed Time: {round(time.time() - start, 3)}s")

File: BOJ/nodejs/2003.js
/*
수들의 합 2
boj.kr/2003
 */
const start = new Date().getTime();

const fs = require("fs");
const input = fs.readFileSync("input.txt").toString().split("\n");
//const input = fs.readFileSync("/dev/stdin").toString().split("\n");

const [N, M] = input[0].split(" ").map(Number);
const A = input[1].split(" ").map(Number);

let ans = 0,
  sum = 0,
  lt = 0;
for (let rt = 0; rt < N; rt++) {
  sum += A[rt];
  if (sum === M) ans++;

  while (sum >= M) {
    sum -= A[lt++];
    if (sum === M) {
      ans++;
    }
  }
}

console.log(ans);

const end = new Date().getTime();
console.log("--------------------");
console.log(`Elapsed Time: ${(new Date().getTime() - start) / 1000}s`);
